packages/headless/src/controllers/instant-results/instant-results.ts

```diff
@@ -0,0 +1,131 @@
+import {SearchEngine} from '../../app/search-engine/search-engine';
+import {Result} from '../../api/search/search/result';
+import {InstantResultSection} from '../../state/state-sections';
+import {buildController, Controller} from '../controller/headless-controller';
+import {instantResults} from '../../app/reducers';
+import {loadReducerError} from '../../utils/errors';
+import {validateOptions} from '../../utils/validate-payload';
+import {
+  fetchInstantResults,
+  registerInstantResults,
+  updateInstantResultsQuery,
+} from '../../features/instant-results/instant-results-actions';
+import {
+  InstantResultOptions,
+  instantResultsOptionsSchema,
+} from './instant-results-options';
+import {SearchAPIErrorWithStatusCode} from '../../api/search/search-api-error-response';
+import {SerializedError} from '@reduxjs/toolkit';
+
+export interface InstantResultProps {
+  options: InstantResultOptions;
+}
+
+export interface InstantResults extends Controller {
+  /**
+   * Updates the instant results query and shows instant results for it.
+   *
+   * @param q - The string value to get instant results for.
```

Comment:
One thing that seems weird is the `q` keyword used throughout this PR, or `updateQuery`...

Would it make more sense to use the keyword "suggestion" instead ? 

While it's true you could use any query to get instant result from, we might want to encourage/guide implementers to use query suggestions instead (which is a bit more precise/constrained), in order to get good instant results.

Reply:
After our conversation last week, it feels to me like we're not coupling instant results to suggestions so tightly, so it feels like it should still be called query, what do you think?

Reply:
Alright !